### PR TITLE
Add section on code review to deployment process docs.

### PR DESCRIPTION
**Pull request by tadhg-ohiggins**
We review the code
Really. We actually do.
It better be good!

Adding a section outlining how we do code review, so that this can be referred to from the SSP. (It's also useful to have it written down here.)

### CODE REVIEW

File: docs/deployment-process.md [hunk 1]
```diff
@@ -121,6 +121,24 @@ cloud.gov UAA application for its users.
 Tock uses the cloud.gov service account service to provide deployer accounts for
 staging and production environments.
 
+### Code review
+
+Submissions to the Tock codebase are made via GitHub, and are only accepted into the main
+branch after review. At least one approving review is required before a branch is merged
+to main, and this restriction is enforced by Tock's GitHub settings.
```
Comment (tbaxter-18f) [inline docs/deployment-process.md:128]:
```suggestion
to main, and this restriction is enforced by Tock's GitHub settings. Your reviewer will usually merge the branch for you.
```

Comment (tbaxter-18f) [reply docs/deployment-process.md:128]: this has been a question that's come up more than once

File: docs/deployment-process.md [hunk 2]
```diff
@@ -121,6 +121,24 @@ cloud.gov UAA application for its users.
...
+
+Code review covers both code (such as Python or JavaScript) and configuration (such as
+Dockerfiles, CloudFoundry manifest files, etc.).
+
+Code review includes assessment of:
```
Comment (tbaxter-18f) [inline docs/deployment-process.md:133]:
```suggestion
Code reviews should be conducted [following the 18F Engineering Guide](https://engineering.18f.gov/code-review/) and include an assessment of:
```

Comment (tbaxter-18f) [reply docs/deployment-process.md:133]: Just cause I really like how the Eng guide spells out how to conduct good code reviews. Feel free to rephrase or reject.

Comment (tadhg-ohiggins) [reply docs/deployment-process.md:133]: Good suggestion!

**Review by tbaxter-18f — APPROVED** [review]

File: docs/deployment-process.md [hunk 3]
```diff
@@ -121,6 +121,25 @@ cloud.gov UAA application for its users.
...
+Your reviewer will usually merge the branch for you.
+
+Code review covers both code (such as Python or JavaScript) and configuration (such as
+Dockerfiles, CloudFoundry manifest files, etc.).
+
+Code reviews should be conducted [following the 18F Engineering Guide](https://engineering.18f.gov/code-review/) and include an assessment of:
+
+- Simplicity. Ideally the submission implements the feature/fix with as little complexity as possible.
+- Legibility. Ideally the submission is easy to understand.
+- Security. The submission is reviews for security considerations.
```
Comment (Jkrzy) [inline docs/deployment-process.md:138]:
```suggestion
- Security. The submission is reviewed for security considerations.
```

**Pull request merged**